### DISCUSSION TITLE
Factorization of rational functions

### DISCUSSION THREAD

**Discussion by ghalimi** [stdlib-js/stdlib#380] [Q&A] [answered]
Some functions use the same rational functions. For example, [erfinv](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/erfinv) and [erfcinv](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/erfcinv) both use the same [rational_p1q1](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/erfinv/lib/rational_p1q1.js), albeit through duplicate auto-generated folders. But [besselj](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/besselj0) uses a totally different yet homonymous [rational_p1q1](https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/math/base/special/besselj0/lib/rational_p1q1.js). Shouldn't these homonymous functions be named differently so that factorization could be implemented in a robust fashion in the reference codebase? Or are there complex build-time mechanisms upstream that already take care of this?

**Comment by kgryte** [comment, marked as answer]
Using different names is unnecessary, as their usage is scoped to the relative module scope. Should these functions need to be present in the same namespace, similar to a C compiler, one can rename symbols when inserting into a common namespace. We do this already when generating "flat" bundles in which all functions exist in the same top-level scope.
